### PR TITLE
feat: add a config to enable retina quality images in screenshots

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
Adding support for configuring image quality for screenshots in both Chrome and Firefox. 

### TESTING INSTRUCTIONS
Update the config override with a pixel_density of 2. Create a report with an email and check the filesize for a 2x size. It should still be captured at the same width as declared in the config, but at a higher resolution. The email will also display the image at 1000px as per the html. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -409,12 +410,13 @@ def _try_json_readsha(filepath: str, length: int) -> Optional[str]:
     # This could cause the server to run out of memory or compute.
     "ALLOW_FULL_CSV_EXPORT": False,
     "UX_BETA": False,
+    "SCREENSHOTS_USE_RETINA_HIRES": False
```
Comment (eschutho) [inline superset/config.py:413]: this is the only change.. the rest is automated linting.

Comment (betodealmeida) [reply superset/config.py:413]: Nit, trailing comma:

```suggestion
    "SCREENSHOTS_USE_RETINA_HIRES": False,
```

**Review by betodealmeida — APPROVED** [review]
LGTM, but what's the workflow here? Doesn't this depend on the device that's loading the screenshot? Is there a way to send both resolutions (1x and 2x) in a report and let the device choose one?

**Comment by eschutho** [comment]
> LGTM, but what's the workflow here? Doesn't this depend on the device that's loading the screenshot? Is there a way to send both resolutions (1x and 2x) in a report and let the device choose one?

Yeah, good question, so I'm using the term retina very loosely. It's basically a higher res image. The rendered image size in the html is 1000px, and the slice for example defaults at 3000px wide image capture, which will place a 3x resolution image in the html. For people who are trying to read the image in the email itself, the text will likely be so small that it's unreadable, so they'll opt to open the image as an attachment and zoom in. 

If you want to take a screenshot at a lower window size so that you can maximize text readability, but still allow people to zoom into the image (some email applications allow you to zoom in in the html and some force you to download the attachment, while some don't let you zoom in at all) then you would set your `WEBDRIVER_WINDOW` config to something smaller like 1400px or 1700px and then increase the screen capture resolution (2x). 

Another option instead of a feature flag is to make this a config and set it at 1x as a default and allow people to set the ratio themselves. At 2x we're seeing fair quality results but as devices get more pixels (I think we may be up to 4x now) we may want to increase the resolution.

File: superset/config.py [hunk 2]
```diff
@@ -1084,7 +1094,8 @@ def SQL_QUERY_MUTATOR(  # pylint: disable=invalid-name,unused-argument
 WEBDRIVER_TYPE = "firefox"
 
 # Window size - this will impact the rendering of the data
-WEBDRIVER_WINDOW = {"dashboard": (1600, 2000), "slice": (3000, 1200)}
+WEBDRIVER_WINDOW = {"dashboard": (
+    1600, 2000), "slice": (3000, 1200), "pixel_density": 1}
```
Comment (eschutho) [inline superset/config.py:1098]: this is the only manual change.. the rest is automated linting.

**Comment by eschutho** [comment]
So after leaving the last comment, I convinced myself to go with the config route, b/c it will be more flexible.

**Review by betodealmeida — APPROVED** [review]
This is great!

**Comment by sadpandajoe** [comment]
🏷️ 2021.46

**Comment by eschutho** [comment]
This fixes a portion of https://github.com/apache/superset/issues/13954

**Pull request merged**